provider-directory/src/main/java/gov/va/api/health/providerdirectory/service/controller/location/LocationController.java

```diff
@@ -1,3 +1,455 @@
 package gov.va.api.health.providerdirectory.service.controller.location;
 
-public class LocationController {}
+import gov.va.api.health.providerdirectory.service.CareSitesResponse;
+import gov.va.api.health.providerdirectory.service.CountParameter;
+import gov.va.api.health.providerdirectory.service.ProviderResponse;
+import gov.va.api.health.providerdirectory.service.ProviderServicesResponse;
+import gov.va.api.health.providerdirectory.service.client.PpmsClient;
+import gov.va.api.health.providerdirectory.service.controller.Bundler;
+import gov.va.api.health.providerdirectory.service.controller.Bundler.BundleContext;
+import gov.va.api.health.providerdirectory.service.controller.PageLinks.LinkConfig;
+import gov.va.api.health.providerdirectory.service.controller.Parameters;
+import gov.va.api.health.providerdirectory.service.controller.Validator;
+import gov.va.api.health.stu3.api.resources.Location;
+import gov.va.api.health.stu3.api.resources.OperationOutcome;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import javax.validation.constraints.Min;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * Request Mappings for the Location Profile, see
+ * http://www.fhir.org/guides/argonaut/pd/StructureDefinition-argo-location.html for implementation
+ * details.
+ */
+@SuppressWarnings("WeakerAccess")
+@RestController
+@RequestMapping(
+  value = {"/api/Location"},
+  produces = {"application/json", "application/json+fhir", "application/fhir+json"}
+)
+public class LocationController {
+
+  private Transformer transformer;
+
+  private PpmsClient ppmsClient;
+
+  private Bundler bundler;
+
+  private int totalRecords;
```

Comment:
`totalRecords` should not be a variable on this class. I don't know if Spring uses the same controller in multiple threads at runtime, but this would not be threadsafe. I think you can have the `search` method return `Pair<List<LocationWrapper>, Integer>` to capture the results and the total number of records.